app/models/concerns/curation_concerns/file_set/derivatives.rb

```diff
@@ -0,0 +1,71 @@
+module CurationConcerns
+  module FileSet
+    module Derivatives
+      extend ActiveSupport::Concern
+
+      included do
+        Hydra::Derivatives.source_file_service = CurationConcerns::LocalFileService
+        Hydra::Derivatives.output_file_service = CurationConcerns::PersistDerivatives
+        Hydra::Derivatives::FullTextExtract.output_file_service = CurationConcerns::PersistDirectlyContainedOutputFileService
+        after_destroy :cleanup_derivatives
+      end
+
+      # This completely overrides the version in Hydra::Works so that we
+      # read and write to a local file. It's important that characterization runs
+      # before derivatives so that we have a credible mime_type field to work with.
+      def create_derivatives(filename)
```

Comment:
Can you extend CC::FileSet::Derivatives and avoid having to duplicate the non-geo formats below?